File: app/src/main/java/com/pr0gramm/app/services/TrackingModule.java
package com.pr0gramm.app.services;

import android.app.Application;

import com.google.firebase.analytics.FirebaseAnalytics;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 */
@Module
public class TrackingModule {
    @Provides
    @Singleton
    public FirebaseAnalytics firebaseAnalyticsTracker(Application app) {
        return FirebaseAnalytics.getInstance(app);
    }
}

File: app/src/main/java/com/pr0gramm/app/TrackingModule.kt
package com.pr0gramm.app

import android.app.Application
import com.google.firebase.analytics.FirebaseAnalytics
import org.kodein.di.Kodein
import org.kodein.di.erased.bind
import org.kodein.di.erased.instance

/**
 */
fun trackingModule(app: Application) = Kodein.Module("tracking") {
    bind<FirebaseAnalytics>() with instance(FirebaseAnalytics.getInstance(app).apply {
        setAnalyticsCollectionEnabled(true)
    })
}
